### PR TITLE
fix(amazonq): handle proxy-related error cases

### PR DESCRIPTION
**Pull request by dhasani23**
## Problem

We want to give users more helpful error message for proxy-related errors they may experience.

## Solution

Show users pre-approved text when these proxy-related errors happen.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/amazonqGumby/chat/controller/controller.ts [hunk 1]
```diff
@@ -435,10 +435,8 @@ export class GumbyController {
             this.messenger.sendKnownErrorResponse('no-alternate-dependencies-found', message.tabID)
             await this.continueTransformationWithoutHIL(message)
         } else if (message.error instanceof ModuleUploadError) {
-            this.messenger.sendUnrecoverableErrorResponse('upload-to-s3-failed', message.tabID)
```
Comment (dhasani23) [inline packages/core/src/amazonqGumby/chat/controller/controller.ts:438]: This and the one below were causing duplicate chat messages to show.

**Review by tincheng — APPROVED** [review]

File: packages/core/src/codewhisperer/service/transformByQ/transformApiHandler.ts [hunk 2]
```diff
@@ -136,6 +136,10 @@ export async function uploadArtifactToS3(
         let errorMessage = `The upload failed due to: ${(e as Error).message}`
         if (errorMessage.includes('Request has expired')) {
             errorMessage = CodeWhispererConstants.errorUploadingWithExpiredUrl
+        } else if (errorMessage.includes('Failed to establish a socket connection')) {
```
Comment (justinmk3) [inline packages/core/src/codewhisperer/service/transformByQ/transformApiHandler.ts:139]: is there a `(e as Error).code` that can be checked ?

Comment (damntrecky) [reply packages/core/src/codewhisperer/service/transformByQ/transformApiHandler.ts:139]: +1 to this.

**Review by damntrecky — APPROVED** [review]

File: packages/core/src/codewhisperer/service/transformByQ/transformApiHandler.ts (hunk 2 above)
Comment (dhasani23) [reply packages/core/src/codewhisperer/service/transformByQ/transformApiHandler.ts:139]: image
The `code` field doesn't appear to be part of the `Error` interface, and `name` would be too generic.

**Pull request merged**